validator/client/validator.go

```diff
@@ -1146,37 +1141,48 @@ func (v *validator) checkDependentRoots(ctx context.Context, head *structs.HeadE
 	if head == nil {
 		return errors.New("received empty head event")
 	}
-	if v.duties == nil {
-		return errors.New("duties are not initialized")
-	}
 	prevDepedentRoot, err := bytesutil.DecodeHexWithLength(head.PreviousDutyDependentRoot, fieldparams.RootLength)
 	if err != nil {
 		return errors.Wrap(err, "failed to decode previous duty dependent root")
 	}
 	if bytes.Equal(prevDepedentRoot, params.BeaconConfig().ZeroHash[:]) {
 		return nil
 	}
-	if !bytes.Equal(prevDepedentRoot, v.duties.PrevDependentRoot) {
-		if err := v.UpdateDuties(ctx); err != nil {
+	epoch := slots.ToEpoch(slots.CurrentSlot(v.genesisTime) + 1)
+	ss, err := slots.EpochStart(epoch + 1)
+	if err != nil {
+		return errors.Wrap(err, "failed to get epoch start")
+	}
+	deadline := v.SlotDeadline(ss - 1)
+	dutiesCtx, cancel := context.WithDeadline(ctx, deadline)
+	defer cancel()
+	v.dutiesLock.RLock()
+	if v.duties == nil || !bytes.Equal(prevDepedentRoot, v.duties.PrevDependentRoot) {
+		v.dutiesLock.RUnlock()
+		if err := v.UpdateDuties(dutiesCtx); err != nil {
```

Comment:
There's an edge case here if the initial `UpdateDuties`  at node startup hasn' t finished and we receive a new block it will try to update duties again, this will lock the duties again and again until it finishes computing them

Reply:
should we add a comment here or something? how would we know...